src/utils/AddressUtils.ts

```diff
@@ -68,3 +70,170 @@ export function isValidEvmAddress(address: string): boolean {
+    return utils.hexZeroPad(utils.hexlify(this.rawAddress), 32).toLowerCase();
+  }
+
+  // Converts the address (can be bytes32 or bytes20) to its base58 counterpart. This conversion will always succeed, even if the input address is not valid on Solana,
+  // as this address may be needed to represent an EVM address on Solana.
+  toBase58(): string {
+    return bs58.encode(this.rawAddress);
+  }
+
+  // Converts the address to a valid EVM address. If it is unable to convert the address to a valid EVM address for some reason, such as if this address
+  // is longer than 20 bytes, then this function will throw an error.
+  toAddress(): string {
+    const hexString = utils.hexlify(this.rawAddress);
+    const rawAddress = utils.hexZeroPad(utils.hexStripZeros(hexString), 20);
+    return utils.getAddress(rawAddress);
+  }
+
+  // Implements `Hexable` for `Address`. Needed for encoding purposes. This class is treated by default as a bytes32 primitive type, but can change for subclasses.
+  toHexString(): string {
+    return this.toBytes32();
+  }
+
+  // Checks if this address can be coerced into a bytes20 evm address. Returns true if it is possible and false otherwise.
+  isValidEvmAddress(): boolean {
+    try {
+      this.toAddress();
+      return true;
+    } catch {
+      return false;
+    }
+  }
+
+  // Checks if this address can be coerced into a valid Solana address. Return true if possible and false otherwise.
+  isValidSvmAddress(): boolean {
+    return isSvmAddress(this.toBase58());
+  }
+
+  // Checks if the address is valid on the given chain ID.
+  isValidOn(chainId: number): boolean {
+    if (chainIsEvm(chainId)) {
+      return this.isValidEvmAddress();
+    }
+    return this.isValidSvmAddress();
+  }
+
+  // Checks if the object is an address by looking at whether it has an Address constructor.
+  static isAddress(obj: Record<string, unknown>): boolean {
+    return obj instanceof Address;
+  }
+
+  // Converts the input address to a 32-byte hex data string.
+  toString(): string {
+    return this.toHexString();
+  }
+
+  // Checks if the address is the zero address.
+  isZeroAddress(): boolean {
+    return utils.stripZeros(this.rawAddress).length === 0;
+  }
+
+  // Checks if the other address is equivalent to this address.
+  eq(other: Address): boolean {
+    return this.toString() === other.toString();
+  }
+
+  /**
+   * Checks if a contract is deployed at the given address
+   * @param provider A valid Ethers.js provider
+   * @returns A boolean indicating if a contract is deployed at the given address or not (true = contract, false = no contract)
+   */
+  async isContractDeployedToAddress(provider: providers.Provider): Promise<boolean> {
+    if (this.isValidEvmAddress()) {
+      const code = await provider.getCode(this.toAddress());
+      // If the code is not empty, then there is a contract at this address
+      return code !== "0x";
+    }
+    // @todo Make this work for SVM
+    return false;
+  }
```

Comment:
This feels like it could belong outside of the Address class - especially because it adds a dependency on the Provider class. wdyt about factoring it out to a standalone AddressUtils function?

Reply:
I just removed it. I originally had it here since I wanted to move all the address utils functions to the class itself, but since we aren't migrating those other functions yet, I think it's fine to just remove here. https://github.com/across-protocol/sdk/pull/909/commits/92539941d0dcc6c8a974822b9c84d5c5e524a465